app/controllers/katello/api/v2/flatpak_remotes_controller.rb

```diff
@@ -0,0 +1,83 @@
+module Katello
+  class Api::V2::FlatpakRemotesController < Katello::Api::V2::ApiController
+    include Katello::Concerns::FilteredAutoCompleteSearch
+
+    before_action :find_authorized_katello_resource, :except => [:index, :create, :scan, :auto_complete_search]
+    before_action :find_optional_organization, :only => [:index, :auto_complete_search]
+
+    resource_description do
+      name 'Flatpak Remotes'
+    end
+
+    def_param_group :flatpak_remote do
+      param :description, String, :desc => N_("Description of the flatpak remote"), :required => false
+      param :username, String, :desc => N_("Username for the flatpak remote"), :required => false
+      param :token, String, :desc => N_("Token/password for the flatpak remote"), :required => false
+    end
+
+    api :GET, "/organizations/:organization_id/flatpak_remotes", N_("List flatpak remotes")
+    api :GET, "/flatpak_remotes", N_("List flatpak remotes")
+    param :organization_id, :number, :desc => N_("organization identifier"), :required => false
+    param :name, String, :desc => N_("Name of the flatpak remote"), :required => false
+    param :label, String, :desc => N_("Label of the flatpak remote"), :required => false
+    param_group :search, Api::V2::ApiController
+    add_scoped_search_description_for(FlatpakRemote)
+    def index
+      respond(:collection => scoped_search(index_relation, :name, :asc))
+    end
+
+    def index_relation
+      remotes = FlatpakRemote.readable
+      remotes = remotes.where(organization_id: @organization.id) if @organization
+      remotes = remotes.where(name: params[:name]) if params[:name]
+      remotes = remotes.where(label: params[:label]) if params[:label]
+      remotes
+    end
+
+    api :GET, "/flatpak_remotes/:id", N_("Show a content view")
+    param :id, :number, :desc => N_("Flatpak remote numeric identifier"), :required => true
+    def show
+      respond :resource => @flatpak_remote
+    end
+
+    api :POST, "/flatpak_remote", N_("Create a flatpak remote")
+    param :name, String, :desc => N_("name"), :required => true
+    param :url, String, :desc => N_("url"), :required => true
+    param :organization_id, :number, :desc => N_("organization identifier"), :required => true
+    param_group :flatpak_remote
+    def create
+      flatpak_remote = FlatpakRemote.new(flatpak_remote_params)
+      respond_for_create(resource: flatpak_remote)
```

Comment:
I'm getting a return like this from cURL:
```
{"name":"fedora-registry2","url":"https://registry.fedoraproject.org/","description":null,"username":null,"token":null,"seeded":false,"registry_url":null,"id":null,"organization_id":1,"organization":{"name":"Default Organization","label":"Default_Organization","id":1}}
```
It looks like the API is just returning the inputs?

Reply:
I think we need a `flatpak_remote.save!` here.